src/scripts/modules/add_destination.js

```diff
@@ -0,0 +1,130 @@
+import destinationsList from "../destinationList.js";
+import events from "./pubsub.js";
+
+let form;
+let uploadBtn;
+let radioBtns;
+let radioVisited;
+let radioBucketlist;
+let cityInput;
+let countryInput;
+let photoInput;
+let newCity;
+let newCountry;
+let newPhoto;
+// let updatedList
+
+function cacheDom() {
+  form = document.getElementById("add-city-form");
+  uploadBtn = document.getElementById("upload");
+  radioBtns = document.querySelectorAll("input[type=radio]");
+  radioVisited = document.getElementById("rd-visited");
+  radioBucketlist = document.getElementById("rd-bucketlist");
+  cityInput = document.getElementById("inputCity");
+  countryInput = document.getElementById("inputCountry");
+  photoInput = document.getElementById("inputImg");
+}
+
+function bindEvents() {
+  form.addEventListener("submit", (event) => {
+    setValues();
+    addDestination();
+    event.preventDefault();
+    console.log(destinationsList);
+  });
+
+  // events.publish('destinationAdded', (form.addEventListener("submit", (event) => {
+  //   setValues();
+  //   addDestination();
+  //   event.preventDefault();
+  //   console.log(destinationsList);
+  // })));
+
+  // updatedList = form.addEventListener("submit", (event) => {
+  //   setValues();
+  //   addDestination();
+  //   event.preventDefault();
+  //   console.log(destinationsList);
+  //   events.publish('destinationAdded', updatedList)
+  // });
+
+  radioVisited.addEventListener("click", isVisited);
+  radioBucketlist.addEventListener("click", isBucketlist);
+  cityInput.addEventListener("change", (event) => {
+    updateCity("city", event.target.value);
+  });
+  countryInput.addEventListener("change", (event) => {
+    updateCountry("country", event.target.value);
+  });
+  photoInput.addEventListener("change", (event) => {
+    updatePhoto("photo", event.target.value);
+  });
+}
+
+function setValues() {
+  return {
+    city: cityInput,
+    country: countryInput,
+    visited: "",
+    url: photoInput,
+  };
+}
+
+function addDestination() {
+  return destinationsList.push({
+    ...newCity,
+    ...newCountry,
+    ...newPhoto,
+  });
+}
+function isVisited() {
+  if (radioVisited.checked === true) {
+    newCity.visited = true;
+  }
+}
+
+function isBucketlist() {
+  if (radioBucketlist.checked === true) {
+    newCity.visited = false;
+  }
+}
+
+function initRadioBtn() {
+  radioBtns.forEach((button) => {
+    isVisited();
+    isBucketlist();
+  });
+}
+
+function updateCountry(fieldName, value) {
+  console.log("updateCountry:", fieldName, value);
+  newCountry = {
+    ...newCountry,
+    [fieldName]: value,
+  };
+}
```

Comment:
I think I wasn't clear the way you can reuse this function to all the 3 fields. 

You can just create one const `formState` that stores all the form fields values in one object, and mutate/reuse it across the different fields.

```js
let formState = {
  city: null,
  country: null,
  photo: null,
}

// ...

  cityInput.addEventListener("change", (event) => {
    updateFormState("city", event.target.value);
  });
  countryInput.addEventListener("change", (event) => {
    updateFormState("country", event.target.value);
  });
  photoInput.addEventListener("change", (event) => {
    updateFormState("photo", event.target.value);
  });

// ....

function updateFormState(fieldName, value) {
  console.log("updateFormState:", fieldName, value);
  formState = {
    ...formState,
    [fieldName]: value,
  };
}

// ....

function addDestination() {
  console.log('addDestination:', formState) // the form with all the field values updated.
  return destinationsList.push(formState);
}
```